### PR TITLE
[Dashing backport] Fix high CPU - Use executor to spin and stop node in tf_listener thread

### PR DESCRIPTION
**Pull request by rotu**
Reduce high CPU usage in TransformListener, and do it in a way that preserves ABI compatibility.
This is a backport of #119.

1. Use the executor `spin` method, so the executor can idle when it has no work.
2. Capture the shared executor so we can (synchronously) call `cancel` in the `unique_ptr<thread>` deleter instead of polling `stop_thread_` in the thread run.

### CODE REVIEW

**Comment by clalancette** [comment]
@rotu @cottsay OK, so I see where you are going with this.  What I'd like to do is get #119 finished and merged onto ros2 (which should be an ABI break; see https://github.com/ros2/geometry2/pull/119#discussion_r306484748).  Once that is done, we can then consider this one as a backport for dashing patch release 2 (the only difference being that we don't remove the now-vestigial `stop_thread` member variable).

File: tf2_ros/src/transform_listener.cpp [hunk 1]
```diff
@@ -60,19 +60,28 @@ TransformListener::~TransformListener()
...
     };
+
   dedicated_listener_thread_ = thread_ptr(
     new std::thread(run_func, node_base_interface),
-    [](std::thread * t) {
+    [executor](std::thread * t) {
+      executor->cancel();
       t->join();
       delete t;
+
```
Comment (cottsay) [inline tf2_ros/src/transform_listener.cpp]: These unrelated whitespace changes should be reverted - especially in a backport.

**Review by cottsay — APPROVED** [review]

**Review by clalancette — APPROVED** [review]
